src/pubsub/core/callables.py

```diff
@@ -144,63 +149,73 @@ def __init__(self, func: UserListener, firstArgIdx: int, ignoreArgs: Sequence[st
             self.autoTopicArgName = None.
         """
 
-        (allParams, varParamName, varOptParamName, defaultVals) = getargspec(func)
-        if defaultVals is None:
-            defaultVals = []
-        else:
-            defaultVals = list(defaultVals)
+        args, varParamName, varOptParamName, argsDefaults, kwargs, kwargsDefaults, annotations = inspect.getfullargspec(func)
+        self.allArgs = {}
+
+        if (argsDefaults != None):
+            argsDefaults_startsAt = len(args) - len(argsDefaults) - 1
+        for i, variable in enumerate(args):
+            if ((i == 0) and (firstArgIdx > 0)):
+                continue #skip self
+
+            if ((argsDefaults == None) or (i < argsDefaults_startsAt)):
+                self.allArgs[variable] = self.NO_DEFAULT()
```

Comment:
Am I missing something?  Where does `self.NO_DEFAULT` come from?  I only see it defined at the module level.

Reply:
Sorry. I write that code in another file and copied it over. It was formatted a bit different in the old file. I have fixed those typos now though.